benchmarks/benchmarks/sparse.py

```diff
@@ -3,13 +3,13 @@
 """
 from __future__ import division, print_function, absolute_import
 
-import warnings
 import time
 import timeit
 
 import numpy
 import numpy as np
 from numpy import ones, array, asarray, empty, random, zeros
+from scipy._lib._numpy_compat import suppress_warnings
```

Comment:
For the benchmarks, the `suppress_warnings` decorator cannot be used. The benchmarks need to be runnable also against old Scipy versions, and suppress_warnings is a recent addition.

I suggest to not make these changes under `benchmarks/`. The way warnings are ignored should not matter for them.